include/xtensor/xbuilder.hpp

```diff
@@ -738,16 +738,11 @@ namespace xt
 
         std::size_t dim_1 = shape[axis_1];
         std::size_t dim_2 = shape[axis_2];
-
         std::size_t n_dim = arr.dimension();
 
         offset >= 0 ? dim_2 -= offset : dim_1 += offset;
 
         auto diag_size = dim_2 < dim_1 ? dim_2 : dim_1;
-        if (diag_size < 0)
-        {
```

Comment:
This is probably a bug, that diag_size and dim_1 and dim_2 should be integers. otherwise, if offset > dim_2 it will lead to an overflow.
I copied this code more or less from numpy, and they use an signed integer type.